File: 2015/j1/main.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
from pathlib import Path

def read_input(file_name: str):
    """Lit un fichier situé dans le même dossier que le script appelant"""
    path = Path(__file__).resolve().parent / file_name
    with path.open("r", encoding="utf-8") as f:
        return [line.rstrip("\n") for line in f]

def part1():
    input_name = read_input("input.txt")

    stair = 0

    for i in input_name[0]:
        if i == "(":
            stair += 1
        if i == ")":
            stair -=1

    return stair

def part2():
    input_name = read_input("input.txt")

    position = 0
    stair = 0

    for i in input_name[0]:
        position += 1
        if i == "(":
            stair += 1
        if i == ")":
            stair -=1

        if stair == -1:
            return position


def main():
    print(f"Part 1: {part1()}")
    print(f"Part 2: {part2()}")

if __name__ == "__main__":
    main()

File: 2015/j1/main.kt
package aoc

import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.Paths

fun readInput(fileName: String): List<String> {
    val path: Path = Paths.get("").resolve(fileName)
    return Files.readAllLines(path)
}

fun part1(): Any {
    val inputName = readInput("input.txt")[0]

    var stair = 0

    for (i in inputName) {
        if (i.equals('(')) {
            stair += 1
        }
        if (i.equals(')')) {
            stair -= 1
        }
    }

    return stair
}

fun part2(): Any {
    val inputName = readInput("input.txt")[0]

    var stair = 0
    var position = 0

    for (i in inputName) {
        position += 1
        if (i.equals('(')) {
            stair += 1
        }
        if (i.equals(')')) {
            stair -= 1
        }

        if (stair.equals(-1)) {
            break
        }
    }

    return position
}

fun main() {
    println("Part 1: ${part1()}")
    println("Part 2: ${part2()}")
}

